File: src.back/main/kotlin/xyz/icedtech/advAgri/agri/AgriBlock.java
package xyz.icedtech.advAgri.agri;

import net.fabricmc.fabric.api.object.builder.v1.block.FabricBlockSettings;
import net.minecraft.block.Block;
import net.minecraft.block.Material;
import xyz.icedtech.advAgri.agri.block.*;


public class AgriBlock {
    public final Block PHOS_ORE = this.FastAddOreBlock();
    public final CropRacksBlock CROP_RACKS = new CropRacksBlock(FabricBlockSettings.of(Material.BAMBOO).hardness(0.5F).breakByHand(true));
    public final Block QUARRY = this.FastAddSoilBlock();
    public final FermentTankBlock FERMENT_TANK = new FermentTankBlock(FabricBlockSettings.of(Material.STONE).hardness(4).breakByHand(false));
    public final Block SILT = this.FastAddSoilBlock();
    public final Block HUMUS = this.FastAddSoilBlock();
    public final Block ACIDIFIED_SOIL = this.FastAddSoilBlock();
    public final Block HARDENED_SOIL = this.FastAddSoilBlock();
    public final Block SULPHUR_ORE = this.FastAddOreBlock();
    public final Block NETHER_SULPHUR_ORE = this.FastAddOreBlock();
    public final BambooFence BAMBOO_FENCE = new BambooFence(FabricBlockSettings.of(Material.BAMBOO).hardness(2.0F).breakByHand(true));
    public final BambooBlock BAMBOO_BLOCK = new BambooBlock(FabricBlockSettings.of(Material.BAMBOO).hardness(2.0F).breakByHand(true));
    public final CeramicJars CERAMIC_JARS = new CeramicJars(FabricBlockSettings.of(Material.STONE).hardness(2.0F).breakByHand(true));

    public final Block COPPER_ORE = this.FastAddOreBlock();
    public final Block TIN_ORE = this.FastAddOreBlock();
    public final Block ZINC_ORE = this.FastAddOreBlock();
    public final Block LEAD_ORE = this.FastAddOreBlock();
    public final Block TUNGSTEN_ORE = this.FastAddOreBlock();
    public final Block ALUMINIUM_ORE = this.FastAddOreBlock();
    public final Block SILVER_ORE = this.FastAddOreBlock();

    public final Block COPPER_BLOCK = this.FastAddOreBlock();
    public final Block TIN_BLOCK = this.FastAddOreBlock();
    public final Block ZINC_BLOCK = this.FastAddOreBlock();
    public final Block LEAD_BLOCK = this.FastAddOreBlock();
    public final Block TUNGSTEN_BLOCK = this.FastAddOreBlock();
    public final Block ALUMINIUM_BLOCK = this.FastAddOreBlock();
    public final Block SILVER_BLOCK = this.FastAddOreBlock();
    public final Block STEEL_BLOCK = this.FastAddOreBlock();
    public final Block NICKEL_BLOCK = this.FastAddOreBlock();
    public final Block TITANIUM_BLOCK = this.FastAddOreBlock();
    public final Block BRONZE_BLOCK = this.FastAddOreBlock();

    public final HighTemperatureElectrolyticFurnace HIGH_TEMPERATURE_ELECTROLYTIC_FURNACE = new
            HighTemperatureElectrolyticFurnace(FabricBlockSettings.of(Material.STONE).hardness(3.5F).breakByHand(true));
    public final ElementaryWellBlock ELEMENTARY_WELL_BLOCK = new
            ElementaryWellBlock(FabricBlockSettings.of(Material.STONE).hardness(3.5F).breakByHand(false));
    public final PrimaryManagementBox PRIMARY_MANAGEMENT_BLOCK = new
            PrimaryManagementBox(FabricBlockSettings.of(Material.STONE).hardness(3.5F).breakByHand(false));
    public final PrimaryPumps PRIMARY_PUMPS = new
            PrimaryPumps(FabricBlockSettings.of(Material.STONE).hardness(3.5F).breakByHand(false));
    public final Block MACHINE_CHASSIS = new
            Block(FabricBlockSettings.of(Material.STONE).hardness(2.0F).breakByHand(true));

    public final Block ELEMENTARY_FLUID_TANK = new
            Block(FabricBlockSettings.of(Material.STONE).hardness(2.0F).breakByHand(true));

    public final Block ELEMENTARY_SUNLIGHT_HARVESTER = new Block(FabricBlockSettings.of(Material.STONE).hardness(2.0F).breakByHand(false));


    private Block FastAddOreBlock() {
        return new Block(FabricBlockSettings.of(Material.METAL).hardness(4.0f));
    }

    private Block FastAddSoilBlock() {
        return new Block(FabricBlockSettings.of(Material.METAL).hardness(1.0f).breakByHand(true));
    }

}

File: src/main/kotlin/xyz/icedtech/advAgri/agri/AgriBlock.kt
package xyz.icedtech.advAgri.agri

import net.fabricmc.fabric.api.block.FabricBlockSettings.of
import net.minecraft.block.Block
import net.minecraft.block.Material
import xyz.icedtech.advAgri.agri.block.*
import xyz.icedtech.advAgri.material.AgriMaterial


class AgriBlock {
    //磷矿
    private var AM = AgriMaterial()
    var PHOS_ORE =
        Block(of(Material.STONE).hardness(3.5F).breakByHand(false).build())

    //作物架 材质&模型OK
    var CROP_RACKS =
        CropRacksBlock(of(AM.PLANT_CROP).hardness(0.1F).breakByHand(true).nonOpaque().noCollision().build())

    //蛭石土 材质OK
    var QUARRY =
        Block(of(Material.SOIL).hardness(0.7F).breakByHand(true).build())

    //发酵罐 材质&模型OK
    var FERMENT_TANK =
        FermentTankBlock(of(Material.WOOD).hardness(2.0F).breakByHand(true).build())

    //淤泥 材质OK
    var SILT =
        Block(of(Material.STONE).hardness(0.7F).breakByHand(true).build())

    //腐殖质 材质OK
    var HUMUS =
        Block(of(Material.SOIL).hardness(0.7F).breakByHand(true).build())

    //酸化土 材质OK
    var ACIDIFIED_SOIL =
        Block(of(Material.STONE).hardness(1.2F).breakByHand(true).build())

    //硬化土 材质OK
    var HARDENED_SOIL =
        Block(of(Material.STONE).hardness(1.2F).breakByHand(true).build())

    //硫磺矿石 材质OK
    val SULPHUR_ORE =
        Block(of(Material.STONE).hardness(3.5F).breakByHand(false).build())

    //地狱硫磺矿石 材质OK
    val NETHER_SULPHUR_ORE =
        Block(of(Material.STONE).hardness(3.5F).breakByHand(false).build())

    //竹篱笆
    var BAMBOO_FENCE =
        BambooFence(of(Material.BAMBOO).hardness(2.0F).breakByHand(true).build())

    var BAMBOO_BLOCK =
        BambooBlock(of(Material.BAMBOO).hardness(2.0F).breakByHand(true).build())

    var CERAMIC_JARS =
        CeramicJars(of(Material.STONE).hardness(2.0F).breakByHand(true).build())

    var COPPER_ORE = Block(of(Material.STONE).hardness(3.5F).breakByHand(false).build())
    var TUNGSTEN_ORE = Block(of(Material.STONE).hardness(3.5F).breakByHand(false).build())
    var ZINC_ORE = Block(of(Material.STONE).hardness(3.5F).breakByHand(false).build())
    var TIN_ORE = Block(of(Material.STONE).hardness(3.5F).breakByHand(false).build())
    var LEAD_ORE = Block(of(Material.STONE).hardness(3.5F).breakByHand(false).build())
    var NICKEL_ORE = Block(of(Material.STONE).hardness(3.5F).breakByHand(false).build())
    var ALUMINIUM_ORE = Block(of(Material.STONE).hardness(3.5F).breakByHand(false).build())


    var COPPER_BLOCK = Block(of(Material.STONE).hardness(4.0F).breakByHand(false).build())
    var TUNGSTEN_BLOCK = Block(of(Material.STONE).hardness(4.0F).breakByHand(false).build())
    var ZINC_BLOCK = Block(of(Material.STONE).hardness(4.0F).breakByHand(false).build())
    var TIN_BLOCK = Block(of(Material.STONE).hardness(4.0F).breakByHand(false).build())
    var LEAD_BLOCK = Block(of(Material.STONE).hardness(4.0F).breakByHand(false).build())
    var NICKEL_BLOCK = Block(of(Material.STONE).hardness(4.0F).breakByHand(false).build())
    var ALUMINIUM_BLOCK = Block(of(Material.STONE).hardness(3.5F).breakByHand(false).build())


    var HIGH_TEMPERATURE_ELECTROLYTIC_FURNACE =
        HighTemperatureElectrolyticFurnace(of(Material.STONE).hardness(3.5F).breakByHand(true).build())
    var PRIMARY_WELL_BLOCK =
        PrimaryWellBlock(of(Material.STONE).hardness(3.5F).breakByHand(false).build())
    var PRIMARY_MANAGEMENT_BLOCK =
        PrimaryManagementBox(of(Material.STONE).hardness(3.5F).breakByHand(false).build())
    var PRIMARY_PUMPS =
        PrimaryPumps(of(Material.STONE).hardness(3.5F).breakByHand(false).build())
}